pkg/kv/kvserver/uncertainty/compute.go

```diff
@@ -55,16 +58,26 @@ import (
 //    right-hand side.
 // TODO(nvanbenschoten): fix this bug with range merges.
 //
-func ComputeInterval(txn *roachpb.Transaction, status kvserverpb.LeaseStatus) Interval {
-	// Non-transactional requests do not have uncertainty intervals.
-	// TODO(nvanbenschoten): Yet, they should. Fix this.
-	//  See https://github.com/cockroachdb/cockroach/issues/58459.
-	if txn == nil {
-		return Interval{}
+func ComputeInterval(
```

Comment:
Might be worth pointing out that this can return an empty `Interval{}` meaning that the request is explicitly bypassing uncertainty restarts.